### PR TITLE
Fix clippy lints and regen

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: glib/src/byte_array.rs [hunk 1]
```diff
@@ -187,27 +187,31 @@ impl fmt::Debug for ByteArray {
 
 macro_rules! impl_cmp {
     ($lhs:ty, $rhs: ty) => {
+        #[allow(clippy::redundant_slicing)]
         impl<'a, 'b> PartialEq<$rhs> for $lhs {
             #[inline]
             fn eq(&self, other: &$rhs) -> bool {
                 self[..].eq(&other[..])
```
Comment (sdroege) [inline glib/src/byte_array.rs:194]: Why not fix this instead? You don't need the slicing here

File: glib/src/subclass/interface.rs [hunk 2]
```diff
@@ -47,35 +47,12 @@ macro_rules! prerequisite_list_trait_impl(
     ($($name:ident),+) => (
         impl<$($name: crate::ObjectType),+> PrerequisiteList for ( $($name),+ ) {
             fn types() -> Vec<ffi::GType> {
-                let mut types = Vec::new();
-                prerequisite_list_trait_inner!(types, $($name)+)
+                vec![$($name::static_type().to_glib()),+]
```
Comment (sdroege) [inline glib/src/subclass/interface.rs:50]: Oh... yes that's easier :)

File: glib/src/byte_array.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply glib/src/byte_array.rs:194]: Lazyness mostly.

File: glib/src/subclass/interface.rs (hunk 2 above)
Comment (sdroege) [reply glib/src/subclass/interface.rs:50]: You can do the same simplification in src/subclass/types.rs for the `InterfaceList`

File: glib/src/byte_array.rs (hunk 1 above)
Comment (sdroege) [reply glib/src/byte_array.rs:194]: Removing the `&` and `[..]` is less work than writing `#[allow(clippy::redundant_slicing)]` :rofl:

Comment (GuillaumeGomez) [reply glib/src/byte_array.rs:194]: Without it, we have two cases:

First by removing `[..]`:

```
error[E0277]: can't compare `byte_array::ByteArray` with `&&[u8]`
   --> glib/src/byte_array.rs:193:22
    |
193 |                 self.eq(&other)
    |                      ^^ no implementation for `byte_array::ByteArray == &&[u8]`
...
224 | impl_cmp!(ByteArray, &'a [u8]);
    | ------------------------------- in this macro invocation
    |
    = help: the trait `std::cmp::PartialEq<&&[u8]>` is not implemented for `byte_array::ByteArray`
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: can't compare `byte_array::ByteArray` with `&std::vec::Vec<u8>`
   --> glib/src/byte_array.rs:193:22
    |
193 |                 self.eq(&other)
    |                      ^^ no implementation for `byte_array::ByteArray == &std::vec::Vec<u8>`
...
226 | impl_cmp!(ByteArray, Vec<u8>);
    | ------------------------------ in this macro invocation
    |
    = help: the trait `std::cmp::PartialEq<&std::vec::Vec<u8>>` is not implemented for `byte_array::ByteArray`
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 2 previous errors
```

Second by removing the `&`:

```
error: function cannot return without recursing
   --> glib/src/byte_array.rs:192:13
    |
192 |             fn eq(&self, other: &$rhs) -> bool {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
193 |                 self.eq(other)
    |                 -------------- recursive call site
...
223 | impl_cmp!(ByteArray, [u8]);
    | --------------------------- in this macro invocation
    |
    = note: `-D unconditional-recursion` implied by `-D warnings`
    = help: a `loop` may express intention better if this is on purpose
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: function cannot return without recursing
   --> glib/src/byte_array.rs:192:13
    |
192 |             fn eq(&self, other: &$rhs) -> bool {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
193 |                 self.eq(other)
    |                 -------------- recursive call site
...
224 | impl_cmp!(ByteArray, &'a [u8]);
    | ------------------------------- in this macro invocation
    |
    = help: a `loop` may express intention better if this is on purpose
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: function cannot return without recursing
   --> glib/src/byte_array.rs:192:13
    |
192 |             fn eq(&self, other: &$rhs) -> bool {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
193 |                 self.eq(other)
    |                 -------------- recursive call site
...
225 | impl_cmp!(&'a ByteArray, [u8]);
    | ------------------------------- in this macro invocation
    |
    = help: a `loop` may express intention better if this is on purpose
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: function cannot return without recursing
   --> glib/src/byte_array.rs:192:13
    |
192 |             fn eq(&self, other: &$rhs) -> bool {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
193 |                 self.eq(other)
    |                 -------------- recursive call site
...
226 | impl_cmp!(ByteArray, Vec<u8>);
    | ------------------------------ in this macro invocation
    |
    = help: a `loop` may express intention better if this is on purpose
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: function cannot return without recursing
   --> glib/src/byte_array.rs:192:13
    |
192 |             fn eq(&self, other: &$rhs) -> bool {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
193 |                 self.eq(other)
    |                 -------------- recursive call site
...
227 | impl_cmp!(&'a ByteArray, Vec<u8>);
    | ---------------------------------- in this macro invocation
    |
    = help: a `loop` may express intention better if this is on purpose
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 5 previous errors
```

Comment (sdroege) [reply glib/src/byte_array.rs:194]: `self.as_ref().eq(other)` is what you're looking for or `(&*self).eq(other)`

Comment (sdroege) [reply glib/src/byte_array.rs:194]: If it's actually difficult and I'm just missing something, feel free to ignore me and keep it as it is :)

Comment (GuillaumeGomez) [reply glib/src/byte_array.rs:194]: It's actually one the strangest thing in rust: the auto-dereference isn't always working depending on the context. For example, you can't compare `&&T` with `&T` for some reason. Anyway, merging as is for the moment.

Comment (sdroege) [reply glib/src/byte_array.rs:194]: Ok :)

File: glib/src/subclass/interface.rs (hunk 2 above)
Comment (sdroege) [reply glib/src/subclass/interface.rs:50]: @GuillaumeGomez Are you sending a PR for this one?

Comment (GuillaumeGomez) [reply glib/src/subclass/interface.rs:50]: Oh sorry, didn't see your message. Doing it right away.

**Pull request merged**